core/lib/cmake/ProjectSQLiteCipher.cmake

```diff
@@ -0,0 +1,155 @@
+project(sqlcipher)
+include(ExternalProject)
+
+
+set(OPENSSL_DIR "${CMAKE_CURRENT_SOURCE_DIR}/openssl")
+set(OPENSSL_LIB $<TARGET_FILE:crypto>)
+string(FIND "${CMAKE_OSX_SYSROOT}" "iphone" IS_IOS)
+if(IS_IOS GREATER_EQUAL 0)
+    set(OPENSSL_LIB ${CMAKE_BINARY_DIR}/core/lib/openssl/crypto/${CMAKE_BUILD_TYPE}-${CMAKE_OSX_SYSROOT})
+endif()
+
+set(_ld_flags_ -L${OPENSSL_LIB})
+set(_c_flags_ "-DSQLITE_HAS_CODEC -I${OPENSSL_DIR}/include -L${OPENSSL_LIB}")
+set(_configure_options_ --with-pic --disable-shared --with-crypto-lib=none --disable-tcl --enable-tempstore=yes )
+set(_build_command_ ${CMAKE_MAKE_PROGRAM})
+set(_libs_ -lcrypto)
+set(prefix "${CMAKE_CURRENT_SOURCE_DIR}/sqlcipher")
+
+if(IS_IOS GREATER_EQUAL 0)
+    find_program(XCODE-SELECT xcode-select)
+    if ("${XCODE-SELECT}" MATCHES "-NOTFOUND")
+        message(FATAL_ERROR "The xcode-select program could not be found.")
+    endif()
+
+    execute_process(COMMAND ${XCODE-SELECT} -print-path
+            OUTPUT_VARIABLE XCODE_DEVELOPER_DIR
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+
+    if (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
+        set(OS_COMPILER "iPhoneSimulator")
+        set(_host_ "x86_64-apple-darwin")
+    else()
+        set(OS_COMPILER "iPhoneOS")
+        set(_host_ "arm-apple-darwin")
+        set(SKIP_BUILD_SQLCIPHER ON)
+    endif()
+
+    set(CROSS_TOP "${XCODE_DEVELOPER_DIR}/Platforms/${OS_COMPILER}.platform/Developer")
+    set(CROSS_SDK "${OS_COMPILER}.sdk")
+
+    set(_configure_options_ ${_configure_options_} --host=${_host_})
+    set(_c_flags_ "${_c_flags_} -arch ${CMAKE_OSX_ARCHITECTURES} -isysroot ${CROSS_TOP}/SDKs/${CROSS_SDK}")
```

Comment:
Are you sure this is right way to set C flags?

Reply:
Yes I'm sure about it, do you know a better way to do so ?